model/riscv_stateen_regs.sail

```diff
@@ -0,0 +1,484 @@
+ /*=======================================================================================*/
+/*  This Sail RISC-V architecture model, comprising all files and                        */
+/*  directories except where otherwise noted is subject the BSD                          */
+/*  two-clause license in the LICENSE file.                                              */
+/*                                                                                       */
+/*  SPDX-License-Identifier: BSD-2-Clause                                                */
+/*=======================================================================================*/
+
+function clause currentlyEnabled(Ext_Smstateen) = hartSupports(Ext_Smstateen)
+function clause currentlyEnabled(Ext_Ssstateen) = hartSupports(Ext_Ssstateen)
+
+function enable_mstateen() -> bool = hartSupports(Ext_Smstateen)
+function enable_sstateen() -> bool = currentlyEnabled(Ext_S) & (currentlyEnabled(Ext_Smstateen) | currentlyEnabled(Ext_Ssstateen))
+function enable_hstateen() -> bool = false & (currentlyEnabled(Ext_Smstateen) | currentlyEnabled(Ext_Ssstateen))
```

Comment:
Worth a comment

```
// TODO: Enable when hypervisor is supported.
```

Actually there is `currentlyEnabled(Ext_H)` now so you could maybe just use that.